playpen/repo_level_awareness/vfs/git_vfs.py

```diff
@@ -186,33 +194,45 @@ def reset(self) -> tuple[int, str, str]:
         """
         return self.git(["reset", "--hard", self.git_sha])
 
+    def diff(self, other: "RepoContextSnapshot") -> tuple[int, str, str]:
+        """
+        Returns the diff between the current snapshot and another snapshot.
+        """
+        return self.git(["diff", other.git_sha, self.git_sha])
+
 
 class RepoContextManager:
-    def __init__(self, project_root: Path, llm: BaseChatModel):
+    def __init__(
+        self,
+        project_root: Path,
+        reflection_agent: ReflectionAgent,
+        initial_msg: str | None = None,
+    ):
         self.project_root = project_root
-        self.snapshot = RepoContextSnapshot.initialize(project_root)
-
-        self.reflection_agent = ReflectionAgent(llm=llm, iterations=1, retries=3)
+        self.snapshot = RepoContextSnapshot.initialize(project_root, initial_msg)
+        self.first_snapshot = self.snapshot
+        self.reflection_agent = reflection_agent
 
     def commit(
         self, msg: str | None = None, spawning_result: SpawningResult | None = None
-    ):
+    ) -> bool:
         """
         Commits the current state of the repository and updates the snapshot.
-        Also runs the reflection agent validate the repository state.
+        Also runs the reflection agent validate the repository state. Returns
+        True if the commit was successful, False otherwise.
         """
 
-        reflection_result = AgentResult(encountered_errors=[], modified_files=None)
-        if spawning_result is not None and isinstance(spawning_result, SpawningResult):
-            reflection_result = self.reflection_agent.execute(
-                None, spawning_result.to_reflection_task()
-            )
+        # AgentResult(encountered_errors=[], modified_files=None)
```

Comment:
For the demo this may make sense but I want to turn it back on before we merge IMO.